### PR TITLE
Remove ES6 syntax from examples

### PR DESCRIPTION
**Pull request by autra**
`examples/` are not processed by babel. Therefore they should be written only with ES5 syntax until we drop support from those older browsers.

### CODE REVIEW

File: examples/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        // deactivated rules for es5
+        'no-var': 'off',
+        'prefer-arrow-callback': 'off',
+        'object-shorthand': 'off',
+        'no-param-reassign': ['error', { 'props': false }],
+        'no-mixed-operators': ['error', { allowSamePrecedence: true }],
+
+        // TODO reactivate all the following rules
+        'no-underscore-dangle': 'off',
+
```
Comment (tbroyer) [inline examples/.eslintrc.js]: There's no `"root": true` in this file, so it's merged with the _project global_ `.eslintrc.js`, which means those _overrides_ are useless, and the removed lines here do not _reactivate_ the corresponding rules.

We should probably go for simplicity here and simply override the `parserOptions` and `env`.

File: .eslintrc.js [hunk 2]
```diff
@@ -57,8 +57,11 @@ module.exports = {
 
     // TODO reactivate all the following rules
 
+    // maybe 'no-mixed-operators': ['error', { allowSamePrecedence: true }],
```
Comment (tbroyer) [inline .eslintrc.js:60]: Should this be in another PR?

File: examples/3dtiles.html [hunk 3]
```diff
@@ -42,18 +42,18 @@
...
-            const menuGlobe = new GuiTools('menuDiv', globe, 300);
+            var globe = new itowns.GlobeView(viewerDiv, positionOnGlobe);
+            var menuGlobe = new GuiTools('menuDiv', globe, 300);
 
             itowns.Fetcher.json('layers/JSONLayers/Ortho.json').then(result => globe.addLayer(result));
 
             // function use :
             // For preupdate Layer geomtry :
-            const preUpdateGeo = (context, layer) => {
+            var preUpdateGeo = (context, layer) => {
```
Comment (tbroyer) [inline examples/3dtiles.html]: This lambda needs to be converted into a function.

File: examples/GUI/GuiTools.js [hunk 4]
```diff
@@ -25,58 +25,64 @@ dat.GUI.prototype.hideFolder = function hideFolder(name, value) {
...
+        return seq.indexOf(a.id) < seq.indexOf(b.id);
+    });
+    for (i = 0; i < sortedLayers.length; i++) {
+        this.addImageryLayerGUI(sortedLayers[i]);
     }
 };
 
 GuiTools.prototype.addElevationLayersGUI = function addElevationLayersGUI(layers) {
-    for (var i = 0; i < layers.length; i++) {
+    var i;
```
Comment (tbroyer) [inline examples/GUI/GuiTools.js:84]: Cannot the `var` stay into the `for`?

File: examples/globe.html [hunk 5]
```diff
@@ -38,21 +38,21 @@
...
         <script src="globe.js"></script>
         <script type="text/javascript">
             /* global itowns, document, GuiTools, globeView, promises */
-            const menuGlobe = new GuiTools('menuDiv');
+            var menuGlobe = new GuiTools('menuDiv');
             menuGlobe.view = globeView;
             // Listen for globe full initialisation event
-            globeView.addEventListener(itowns.GLOBE_VIEW_EVENTS.GLOBE_INITIALIZED, () => {
+            globeView.addEventListener(itowns.GLOBE_VIEW_EVENTS.GLOBE_INITIALIZED, function () {
                 // eslint-disable-next-line no-console
```
Comment (tbroyer) [inline examples/globe.html:51]: The examples-specific `.eslintrc.js` could maybe disable the `no-console` rule?

File: webpack.config.js [hunk 6]
```diff
@@ -5,6 +5,11 @@ var definePlugin = new webpack.DefinePlugin({
     __DEBUG__: JSON.stringify(process.env.NODE_ENV === 'development'),
 });
 
+var providePlugin = new webpack.ProvidePlugin({
```
Comment (tbroyer) [inline webpack.config.js:8]: Should we do the same for the promise and fetch polyfills? Or should this be changed to just a new entry in the `itowns` entry point?

File: utils/debug/Debug.js [hunk 7]
```diff
@@ -8,7 +8,7 @@ import View from '../../src/Core/View';
 
 function createChartContainer(chartDiv, width, left, chartId) {
     const div = document.createElement('div');
-    div.style = `position: absolute; bottom: 0; left: ${left}vw; width: ${width}vw; height: 20rem; background-color: white;`;
+    div.style.cssText = `position: absolute; bottom: 0; left: ${left}vw; width: ${width}vw; height: 20rem; background-color: white;`;
```
Comment (tbroyer) [inline utils/debug/Debug.js:11]: Which browser does not support setting `style` directly?

Comment (autra) [reply utils/debug/Debug.js:11]: Some webkit-based, like midori or Epiphany. Safari is or maybe was affected at some point too, if I believe comments on the web (I have no way of testing though)

File: examples/.eslintrc.js (hunk 1 above)
Comment (autra) [reply examples/.eslintrc.js]: >  those overrides are useless

They are not. Try to remove `prefer-arrow-callback` above.

> the removed lines here do not reactivate the corresponding rules.

They do. Try adding a very long line in a js file in examples.

The merging capability is apparently not as clever as you think it is (or I completely missed what you're saying). According to my test, the `rules` property of this file completely shadowed the `rules` property of the main `.eslintrc`.

> We should probably go for simplicity here and simply override the parserOptions and env.

Not enough, according to my tests. Eslint still complains about classic `function` callback until I disabled the `prefer-arrow-callback` above.

File: examples/GUI/GuiTools.js (hunk 4 above)
Comment (autra) [reply examples/GUI/GuiTools.js:84]: We have a linter rule for that. (`var-on-top` or something like this)

It's annoying, but useful too, as this `i` will get hoisted anyway.

File: examples/globe.html (hunk 5 above)
Comment (autra) [reply examples/globe.html:51]: (If you ask me, `no-console` would be disabled everywhere)

Why not?

File: webpack.config.js (hunk 6 above)
Comment (autra) [reply webpack.config.js:8]: I don't know. I haven't touched Promise and fetch because using itowns without either of them is quite hard, so it made sense to me to expose them in window. For TextEncoder and TextDecoder, it's only used internally so I didn't want to expose them needlessly.

Comment (autra) [reply webpack.config.js:8]: So, as I didn't know, I went for the status quo for this PR (already big and painful, I may say! ;-) )

File: examples/3dtiles.html (hunk 3 above)
Comment (autra) [reply examples/3dtiles.html]: Good catch!

File: examples/.eslintrc.js (hunk 1 above)
Comment (tbroyer) [reply examples/.eslintrc.js]: I haven't had much experience with several eslintrc files, and mostly (if not only) redefining only the parser options and environment (and a much less strict ruleset), do I may just have misunderstood the docs or read too much into them.

File: examples/globe.html (hunk 5 above)
Comment (tbroyer) [reply examples/globe.html:51]: To avoid leaving debug logs in production builds?

File: webpack.config.js (hunk 6 above)
Comment (tbroyer) [reply webpack.config.js:8]: With this reasoning, maybe fetch could follow the same pattern, as it's not directly exposed. But let's not do that here anyway.

**Review by tbroyer — COMMENTED** [review]
I believe there's a lambda to turn into a function, but otherwise consider this PR approved.

File: .eslintrc.js (hunk 2 above)
Comment (autra) [reply .eslintrc.js:60]: It's included here, because it matches with the rules in examples/.eslintrc.js

**Comment by autra** [comment]
updated. Fixed a lambda catched by @tbroyer plus another one, and one little indentation mistake.

@iTowns/reviewers please r?

**Review by tbroyer — APPROVED** [review]

File: webpack.config.js (hunk 6 above)
Comment (autra) [reply webpack.config.js:8]: I gave it a thought, and I actually think we shouldn't publicly polyfill *at all*, as we are not an app, but a module. So I think I'll use babel-runtime + babel-plugin-transform-runtime instead, or use polyfill and the ProvidePlugin (not sure which one is best yet).
